pkg/osquery/tables/dataflattentable/flattener_from_parser.go

```diff
@@ -0,0 +1,39 @@
+package dataflattentable
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+
+	"github.com/kolide/launcher/pkg/dataflatten"
+)
+
+type parserInterface interface {
```

Comment:
Looking at this a little more, is there a naming conflict? Can we just call it `parser`? (Reviewing from phone now, I'll check in the morning)

Reply:
checked this morning, looks like `parser` works, my affinity toward using `er` at the end of interfaces comes from the standard lib and [go style guide](https://go.dev/doc/effective_go#interface-names)

Reply:
Hrm. Okay, `parser` it is. I don't think I like it, it feels... too short and ambiguous for an internal interface. I think the go docs make more sense for external exported/common interfaces. For internal ones, I think it's much less clear. 

I tend to favor more words for clarity. But :shrug: